tests/fsharp/Compiler/Language/SlicingQuotationTests.fs

```diff
@@ -0,0 +1,348 @@
+﻿namespace FSharp.Compiler.UnitTests
+
+open NUnit.Framework
+open FSharp.Compiler.SourceCodeServices
+
+[<TestFixture>]
+module SlicingQuotationTests =
```

Comment:
I am wondering if we should make these tests as part of the FSharp.Core unit tests. What do you think?

Reply:
It does not matter much to me. I'm fine with the tests being here. Just want to know your thoughts.

Reply:
I put the tests here because this test is asserting facts about the shape of the rewritten slicing ASTs, and the rewriting happens in typechecker. IIRC the FSharp.Core tests uses typechecker.fs from the bootstrap compiler, so we'd have to test them here.